docs/bugs_and_glitches.md

```diff
@@ -313,119 +311,105 @@ The code in `CheckHiddenOpponent` is completely redundant as `BattleCommand_Chec
 
 ```diff
-.got_mon
-	ld a, [wd002]
-	ld hl, wPartyMonNicknames
-	call GetNick
-	ld a, MON_HP
-	call GetBeatupMonLocation
-	ld a, [hli]
-	or [hl]
-	jp z, .beatup_fail ; fainted
-	ld a, [wd002]
-	ld c, a
-	ld a, [wCurBattleMon]
+ .got_mon
+ 	ld a, [wd002]
+ 	ld hl, wPartyMonNicknames
+ 	call GetNick
+ 	ld a, MON_HP
+ 	call GetBeatupMonLocation
+ 	ld a, [hli]
+ 	or [hl]
+ 	jp z, .beatup_fail ; fainted
+ 	ld a, [wd002]
+ 	ld c, a
+ 	ld a, [wCurBattleMon]
 -	; BUG: this can desynchronize link battles
 -	; Change "cp [hl]" to "cp c" to fix
 -	cp [hl]
 +	cp c
-	ld hl, wBattleMonStatus
-	jr z, .active_mon
-	ld a, MON_STATUS
-	call GetBeatupMonLocation
-.active_mon
-	ld a, [hl]
-	and a
-	jp nz, .beatup_fail
+ 	ld hl, wBattleMonStatus
+ 	jr z, .active_mon
+ 	ld a, MON_STATUS
+ 	call GetBeatupMonLocation
+ .active_mon
+ 	ld a, [hl]
+ 	and a
+ 	jp nz, .beatup_fail
 ```
 
 
 ## Beat Up may fail to raise substitute
 
-*Fixing this bug will break compatibility with standard Pokémon Crystal for link battles.*  
-(Only the fixes denoted with "breaking" will actually break compatibility, the others just affect what's shown on the screen with the patched game)
-
-This is a bug in `BattleCommand_EndLoop` in [engine/battle/effect_commands.asm](/engine/battle/effect_commands.asm) that prevents the rest of the move's effect from being executed if the player or enemy only has one mon in their party while using Beat Up.
-
-It prevents the substitute from being raised and the King's Rock from working.
+*Fixing this bug will break compatibility with standard Pokémon Crystal for link battles.*
 
+This bug prevents the rest of the move's effect from being executed if the player or enemy only has one Pokémon in their party while using Beat Up. It prevents Substitute from being raised and King's Rock from working.
 
-**Fix (breaking):**
+**Fix:** Edit `BattleCommand_EndLoop` in [engine/battle/effect_commands.asm](/engine/battle/effect_commands.asm):
 
 ```diff
-.only_one_beatup
-	ld a, BATTLE_VARS_SUBSTATUS3
-	call GetBattleVarAddr
-	res SUBSTATUS_IN_LOOP, [hl]
+ .only_one_beatup
+ 	ld a, BATTLE_VARS_SUBSTATUS3
+ 	call GetBattleVarAddr
+ 	res SUBSTATUS_IN_LOOP, [hl]
 -	call BattleCommand_BeatUpFailText
 -	jp EndMoveEffect
 +	ret
 ```
 
-**Fix (cosmetics):**
+**Cosmetic fix:** This fix does not break compatibility, but it only affects what's shown on the screen for the patched game.
 
 ```diff
-.only_one_beatup
-	ld a, BATTLE_VARS_SUBSTATUS3
-	call GetBattleVarAddr
-	res SUBSTATUS_IN_LOOP, [hl]
-	call BattleCommand_BeatUpFailText
+ .only_one_beatup
+ 	ld a, BATTLE_VARS_SUBSTATUS3
+ 	call GetBattleVarAddr
+ 	res SUBSTATUS_IN_LOOP, [hl]
+ 	call BattleCommand_BeatUpFailText
 +	call BattleCommand_RaiseSub
-	jp EndMoveEffect
+ 	jp EndMoveEffect
 ```
 
+For the cosmetic fix, also edit `BattleCommand_FailureText` in [engine/battle/effect_commands.asm](/engine/battle/effect_commands.asm).
```

Comment:
This is a completely separate fix, and can (and should) be used in combination with the "breaking" fix above.
Originally, it was stated as simply "Fix:", but in the last commit I thought it'd be nice to specify it won't break compatibility with an original game.

It is, in essence, a different bug in a different place, but it also fits with the "Beat Up may fail to raise substitute", just under different circumstances than the bug above. I'm not entirely sure how to make this crystal clear.